### PR TITLE
Expanding theoretical spectrum generation tutorial

### PR DESCRIPTION
**Pull request by JeeH-K**
by introducing more parameters.
Fixing the issue https://github.com/OpenMS/pyopenms-docs/issues/291

Corresponding visualization (example image) should be changed in different PR after fixing the issue https://github.com/OpenMS/OpenMS/issues/6701

### CODE REVIEW

File: docs/source/theoreticalspectrumgenerator.rst [hunk 1]
```diff
@@ -117,7 +123,18 @@ which you again can visualize with:
 
 The first example shows how to put peaks of a certain type, y-ions in this case, into
 a spectrum. The second spectrum is filled with a complete fragment ion spectrum
-of all peaks (a-, b-, y-ions and losses). The losses are based on commonly
+of all peaks (a-, b-, y-ions, precursor peaks, and losses).
+
+From the peptide with 9 amino acids, each ion type should be produced theoretically
```
Comment (timosachsenberg) [inline docs/source/theoreticalspectrumgenerator.rst:128]: I think "each ion type should be produced theoretically with 8 peaks." might not be clear.

Comment (JeeH-K) [reply docs/source/theoreticalspectrumgenerator.rst:128]: How about:
Here, from the peptide with 9 amino acids, fragments theoretically can occur in 8 different positions, resulting in 8  peaks per ion type (a, b, and y-ion in this example code).

I'm not trying to explain too much, as this is a tutorial, not a lecture note... What fragmentation of a peptide means is out-of-scope for the tutorial, and

Comment (timosachsenberg) [reply docs/source/theoreticalspectrumgenerator.rst:128]: I think that works.
Regarding the lecture part... it is a bit difficult to say where to put such information.
Ideally, we would have it either in https://pyopenms.readthedocs.io/en/latest/introduction.html#background
or in this part (e.g., experienced users can just skip over it but newbies might profit from getting a refresher on ion types etc.).
But yeah that is out of the scope of this sprint.

Comment (JeeH-K) [reply docs/source/theoreticalspectrumgenerator.rst:128]: Agreed. It'd be great if we have a background page to link this

**Review by timosachsenberg — APPROVED** [review]

**Comment by jpfeuffer** [comment]
No merges until CI is fixed.

**Comment by JeeH-K** [comment]
> No merges until CI is fixed.

I'll keep an eye on the technical issues channel on discord then.

**Comment by jpfeuffer** [comment]
Yes, or here: https://github.com/OpenMS/pyopenms-docs/issues/343

**Pull request merged**